meja/typechecker.ml

```diff
@@ -865,6 +865,8 @@ let type_extension ~loc variant ctors env =
   in
   (env, variant, ctors)
 
+let in_decl = ref false
```

Comment:
maybe stick this ref inside of check_statement like

```ocaml
let check_statement =
  let in_decl = ref false in
  let rec check_statement env stmt = ... in
  check_statement

Reply:
The `Codegen.typ_of_decl` call in `check_statement` calls back into `check_statement`, we need it outside to stop the `poly` and `var` declarations from getting their own `poly` and `var` declarations, and so on.